### PR TITLE
Dashboard all/mine filters

### PR DESCRIPTION
**Pull request by Espurna**
## Description of the issue/feature this PR addresses

The ability to change panel dashboards depending on assignation.

## Current behavior before PR

There is no way to update dashboard panels accordingly to user property.

## Desired behavior after PR is merged

A selection list with the values «All» and «Mine» will be added above each section of the Dashboard. The panels and information within each panel will change according to the option selected: for example, if the user selects «Mine», the panel with the summary of worksheets and worksheets statuses will only take into account the worksheets to which the user is assigned.

This feature can be activated/deactivated in bika setup.

selection_028

selection_030

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/subscribers/dashboard_cookies.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    if not is_bika_installed():
+        logger.warn(
+            "Package 'bika.lims' is not installed, skipping event handler "
+            "for IUserLoggedOutEvent.")
+        return
+    request = api.env.getRequest()
+    response = request.RESPONSE
+
+    # Voiding our special cookie on logout
+    response.setCookie(DASHBOARD_FILTER_COOKIE, None,  path='/', max_age=0)
```
Comment (xispa) [inline bika/lims/subscribers/dashboard_cookies.py]: Is this necessary?. I would expect to see the dashboard filtered as I left it some days ago.

File: bika/lims/browser/dashboard/dashboard.py [hunk 2]
```diff
@@ -3,41 +3,109 @@
...
+        cookie_raw = self.request.get(DASHBOARD_FILTER_COOKIE, None)
+        # If it doesn't exist, create it with default values
+        if cookie_raw is None:
+            cookie_raw = self._create_raw_data()
+            self.request.response.setCookie(
+                DASHBOARD_FILTER_COOKIE,
+                json.dumps(cookie_raw),
+                quoted=False,
+                path='/')
+            return cookie_raw
```
Comment (xispa) [inline bika/lims/browser/dashboard/dashboard.py]: This return is not required

**Review by xispa — CHANGES_REQUESTED** [review]
I'd remove the option "DashboardAllMine" from setup and always render the "All/Mine" filter. Note also the following error is rised when the cookie is set for the first time:

```
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module bika.lims.browser.dashboard.dashboard, line 50, in __call__
  Module json, line 338, in loads
  Module json.decoder, line 366, in decode
TypeError: expected string or buffer
```

File: bika/lims/browser/dashboard/dashboard.py [hunk 3]
```diff
@@ -3,41 +3,109 @@
...
             # display the Main Dashboard view
             pm = getToolByName(self.context, "portal_membership")
-            member = pm.getAuthenticatedMember()
-            roles = member.getRoles()
+            self.member = pm.getAuthenticatedMember()
+            roles = self.member.getRoles()
             tofrontpage = 'Manager' not in roles and 'LabManager' not in roles
 
-        if tofrontpage == True:
+        if tofrontpage:
```
[the remark this answers is not in the document]
Comment (Espurna) [reply bika/lims/browser/dashboard/dashboard.py]: https://www.python.org/dev/peps/pep-0008/#id51

File: bika/lims/subscribers/dashboard_cookies.py (hunk 1 above)
Comment (Espurna) [reply bika/lims/subscribers/dashboard_cookies.py]: I was just following what we do with other cookies

**Review by xispa — APPROVED** [review]

**Pull request merged**